bindings/java/src/org/sleuthkit/datamodel/BlackboardArtifact.java

```diff
@@ -193,65 +194,61 @@ public String getDisplayName() {
 	 * @throws TskCoreException if there is a problem creating the description.
 	 */
 	public String getShortDescription() throws TskCoreException {
-		List<BlackboardAttribute> attrs;
-		attrs = new ArrayList<BlackboardAttribute>();  //only allow the adding of one or two items to keep descirption short
+		BlackboardAttribute attr = null;
+		StringBuilder shortDescription = new StringBuilder("");
 		switch (ARTIFACT_TYPE.fromID(artifactTypeId)) {
 			case TSK_WEB_BOOKMARK:  //web_bookmark, web_cookie, web_download, and web_history are the same attribute for now
 			case TSK_WEB_COOKIE:
 			case TSK_WEB_DOWNLOAD:
 			case TSK_WEB_HISTORY:
-				attrs.add(getAttribute(new BlackboardAttribute.Type(ATTRIBUTE_TYPE.TSK_DOMAIN)));
+				attr = getAttribute(new BlackboardAttribute.Type(ATTRIBUTE_TYPE.TSK_DOMAIN));
 				break;
 			case TSK_KEYWORD_HIT:
-				attrs.add(getAttribute(new BlackboardAttribute.Type(ATTRIBUTE_TYPE.TSK_KEYWORD_PREVIEW)));
+				attr = getAttribute(new BlackboardAttribute.Type(ATTRIBUTE_TYPE.TSK_KEYWORD_PREVIEW));
 				break;
 			case TSK_DEVICE_ATTACHED:
-				attrs.add(getAttribute(new BlackboardAttribute.Type(ATTRIBUTE_TYPE.TSK_DEVICE_ID)));
+				attr = getAttribute(new BlackboardAttribute.Type(ATTRIBUTE_TYPE.TSK_DEVICE_ID));
 				break;
 			case TSK_CONTACT: //contact, message, and calllog are the same attributes for now
 			case TSK_MESSAGE:
 			case TSK_CALLLOG:
-				BlackboardAttribute name;
-				if (((name = getAttribute(new BlackboardAttribute.Type(ATTRIBUTE_TYPE.TSK_NAME))) != null) && !name.getDisplayString().isEmpty()) {
-					attrs.add(name);
-				}
-				BlackboardAttribute attr;
-				if ((attr = getAttribute(new BlackboardAttribute.Type(ATTRIBUTE_TYPE.TSK_PHONE_NUMBER))) != null) {
-					attrs.add(attr);
+				//get the first of these attributes which exists and is non null
+				if (((attr = getAttribute(new BlackboardAttribute.Type(ATTRIBUTE_TYPE.TSK_NAME))) != null) && !attr.getDisplayString().isEmpty()) {
```

Comment:
The amount of duplicate code( each if branch is identical except for the attribute type) here makes me nervous,  I feel like a different control structure might be better suited.  What about something like:
```JAVA
<Stream of types> types = { TSK_NAME, TSK_PHONE_NUMBER...}
for (type t: types){
atttr = getAttribute(t);
if (attr != null){
break;
}
}
```